### PR TITLE
Fix bugs in SP report output

### PR DESCRIPTION
**Pull request by awhodgson256**
## What does this pull request do?

Fix bugs with nulls being substituted with current time imestamps

Fix bug with attendance time for first session post SAA (was SAA value)

## What is the intent behind these changes?

Ensure consistency in output for revised SP performance report

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/repository/ReferralPerformanceReportRepositoryImpl.kt [hunk 1]
```diff
@@ -127,9 +127,8 @@ class ReferralPerformanceReportRepositoryImpl : ReferralPerformanceReportReposit
   }
 
   fun timestampToOffset(timestamp: Timestamp?): OffsetDateTime? {
-    val currentTimeMillis = System.currentTimeMillis()
-    val resolved = timestamp ?: Timestamp(currentTimeMillis)
-    return OffsetDateTime.ofInstant(resolved.toInstant(), ZoneId.of("UTC"))
+    val resolved = timestamp
```
Comment (kth13) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/repository/ReferralPerformanceReportRepositoryImpl.kt:130]: I believe you don't need resolved variable. You can directly do `OffsetDateTime.ofInstant(timestamp?.toInstant(), ZoneId.of("UTC"))`.

File: src/main/resources/db/migration/V1_176__update_referral_performance_report_first_attendance_srf.sql [hunk 2]
```diff
@@ -0,0 +1,19 @@
+create or replace function performance_report_first_attendance(referral_id UUID)
```
Comment (kth13) [inline src/main/resources/db/migration/V1_176__update_referral_performance_report_first_attendance_srf.sql:1]: Just out of curiosity, how this function gets used in the main function?

Comment (awhodgson256) [reply src/main/resources/db/migration/V1_176__update_referral_performance_report_first_attendance_srf.sql:1]: It's not called by the main report function @kth13 that's just used by the reader - it gets called at the later stage to find the appointment time for the processor (and added to the output)

**Review by kth13 — APPROVED** [review]
LGTM

**Pull request merged**